contracts/GenArt721CoreV3_Engine.sol

```diff
@@ -0,0 +1,2137 @@
+            "Only artist or Admin ACL allowed"
+        );
+        _;
+    }
+
+    /**
+     * This modifier allows the artist of a project to call a function if the
+     * owner of the contract has renounced ownership. This is to allow the
+     * contract to continue to function if the owner decides to renounce
+     * ownership.
+     */
+    modifier onlyAdminACLOrRenouncedArtist(
+        uint256 _projectId,
+        bytes4 _selector
+    ) {
+        require(
+            adminACLAllowed(msg.sender, address(this), _selector) ||
+                (owner() == address(0) &&
+                    msg.sender ==
+                    projectIdToFinancials[_projectId].artistAddress),
+            "Only Admin ACL allowed, or artist if owner has renounced"
+        );
+        _;
+    }
+
+    /**
+     * @notice Initializes contract.
+     * @param _tokenName Name of token.
+     * @param _tokenSymbol Token symbol.
+     * @param _randomizerContract Randomizer contract.
+     * @param _adminACLContract Address of admin access control contract, to be
+     * set as contract owner.
+     * @param _startingProjectId The initial next project ID.
+     * @param _autoApproveArtistSplitProposals Whether or not to always
+     * auto-approve proposed artist split updates.
+     * @dev _startingProjectId should be set to a value much, much less than
+     * max(uint248), but an explicit input type of `uint248` is used as it is
+     * safer to cast up to `uint256` than it is to cast down for the purposes
+     * of setting `_nextProjectId`.
+     */
+    constructor(
+        string memory _tokenName,
+        string memory _tokenSymbol,
+        address _renderProviderAddress,
+        address _platformProviderAddress,
+        address _randomizerContract,
+        address _adminACLContract,
+        uint248 _startingProjectId,
+        bool _autoApproveArtistSplitProposals
+    )
+        ERC721_PackedHashSeed(_tokenName, _tokenSymbol)
+        onlyNonZeroAddress(_randomizerContract)
+    {
+        // setup immutable `autoApproveArtistSplitProposals` config
+        autoApproveArtistSplitProposals = _autoApproveArtistSplitProposals;
+        // record contracts starting project ID
+        // casting-up is safe
+        startingProjectId = uint256(_startingProjectId);
+        _updateRenderProviderPrimarySalesAddress(_renderProviderAddress);
+        _updateRenderProviderSecondarySalesAddress(_renderProviderAddress);
+        _updatePlatformProviderPrimarySalesAddress(_platformProviderAddress);
+        _updatePlatformProviderSecondarySalesAddress(_platformProviderAddress);
+        _updateRandomizerAddress(_randomizerContract);
+        // set AdminACL management contract as owner
+        _transferOwnership(_adminACLContract);
+        // initialize default base URI
+        _updateDefaultBaseURI(
+            string.concat(
+                "https://token.artblocks.io/",
+                address(this).toHexString(),
```

Comment:
A bit initially confused why this is failing from the error message alone–any thoughts here @ryley-o https://app.circleci.com/pipelines/github/ArtBlocks/artblocks-contracts/1493/workflows/6fb418cb-6a86-4ab1-8584-2790674439dd/jobs/1738?invite=true#step-110-16

Reply:
Aha! Awesome–thank you ser.